src/sysusers/sysusers.c

```diff
@@ -1949,10 +1950,26 @@ static int run(int argc, char *argv[]) {
                 return log_error_errno(errno, "Failed to set SYSTEMD_NSS_BYPASS_SYNTHETIC environment variable: %m");
 
         if (!uid_range) {
-                /* Default to default range of 1..SYSTEM_UID_MAX */
-                r = uid_range_add(&uid_range, &n_uid_range, 1, SYSTEM_UID_MAX);
+                /* Default to default range of SYSTEMD_UID_MIN..SYSTEM_UID_MAX. */
+                login_defs defs;
+                const char *path;
+
+                path = prefix_roota(arg_root, "/etc/login.defs");
+                r = read_login_defs(&defs, path);
```

Comment:
hmm, we have been a bit sloppy with insisting on chase_symlinks() for opening the files in sysusers.c relative to the --root= dir, but I think we should become better at that and at least for new code always use chase_symlinks() so that we don't get confused by /etc/ in the image being a symlink.

In fact we should probably add chase_symlinks_and_fopen(), similar to chase_symlinks_and_open() and use that everywhere. Not doing that for the password databases is actually quite problematic security wise, given that this means invoking the tool on rogue images will possibly manipulate the host password database...

I know this problem is unrelated to the patch at hand, but this has been bugging me for a while, ad for new code I think we should insist on this